### PR TITLE
Revamp light floor tiles

### PR DESCRIPTION
**Pull request by moltoretardo**
# About the pull request
When I initially fixed the light tiles, I found many unused sprite of different colors, decided to add them and change a bit how it works.
Here are the different options:
lights

- Multitools now change the color by clicking on the tile
- Click on them to turn on/off
- Tiles can now be broken by xenos and environment (tested with grenades)
- To repair the floor tile, put a normal light bulb in it 
lightbulb
- Use a crowbar to remove the tile, if it's broken it'll simply destroy it
- Build one by using wire on glass and then one metal (unchanged)
- Their current light emission is the same as a flashlight, may be tweaked eventually to be less
- Currently the broken sprite and the off sprite is the same, should get a broken sprite eventually...
- Most of the old code wasn't used at all or didn't work, rewrote most of it

# Explain why it's good for the game
I think it's a good addition in the toolbelts of Maintenance Technicians, they can decorate rooms and use them for signalization instead of drawing with crayons on the floor. There shouldn't be any balance issues since it's basically a costly flashlight, light bulbs are, as far as I know, only obtainable from the autolathe making it not very reliable. Auto-light replacer doesn't work purposely so it doesn't get weaponized, their purpose is solely to embellish rooms.

It's hardly something that is used, and if it gets abused I'll simply nerf them down to ashes (can't have nice things) 
arroww

# Testing Photographs and Procedure

I've tested everything I could think of, everything should work.
<details>
<summary>Screenshots & Videos</summary>

Nade breaking some of the tiles
beforenade

nade destroy

</details>

# Changelog

:cl: Molto
add: Readded the different light floor colors, use multitool to cycle through them
fix: Fixed light tiles being indestructible unless using a crowbar
code: Moved some part of the code that were only used by light tiles in floors.dm into light.dm directly 
refactor: Made one letter variables more lettered
/:cl:

### CODE REVIEW

File: code/game/objects/items/stacks/tiles/light.dm [hunk 1]
```diff
@@ -8,23 +8,10 @@
...
-		state = 2 //breaking
-	else if(prob(10))
-		state = 1 //flickering occasionally
-	else
-		state = 0 //fine
-
-/obj/item/stack/tile/light/attackby(var/obj/item/O as obj, var/mob/user as mob)
-	..()
-	if(istype(O,/obj/item/tool/crowbar))
+	if(istype(item_in_hand,/obj/item/tool/crowbar))
```
Comment (stanalbatross) [inline code/game/objects/items/stacks/tiles/light.dm:14]: Check for tool trait don't istype

File: code/game/turfs/floor.dm [hunk 2]
```diff
@@ -96,6 +96,7 @@
 	else if(is_light_floor())
 		icon_state = "light_broken"
 		broken = 1
+		SetLuminosity(0)
```
Comment (stanalbatross) [inline code/game/turfs/floor.dm:99]: Does this not have defines

File: code/game/turfs/light.dm [hunk 3]
```diff
@@ -10,33 +10,81 @@
...
+
+/turf/open/floor/light/attack_hand(mob/user as mob) //turning the light on and off
+	if(!broken && on)
+		on = FALSE
+		update_icon()
+		to_chat(user, SPAN_NOTICE("You turn the light off."))
+	else if(!broken && !on)
+		on = TRUE
+		update_icon()
+		to_chat(user, SPAN_NOTICE("You turn the light on."))
```
Comment (stanalbatross) [inline code/game/turfs/light.dm:75]: This can be shorter, use on = !on and use a [on ? "On" : "off"] thing in the tochat

Comment (moltoretardo) [reply code/game/turfs/light.dm]: That's so awesome

File: code/game/turfs/floor.dm (hunk 2 above)
Comment (moltoretardo) [reply code/game/turfs/floor.dm:99]: I haven't found any, I can create them if needed though

Comment (harryob) [reply code/game/turfs/floor.dm:99]: that'd be OOS for this PR, luminosity is a pretty big thing

**Review by harryob — APPROVED** [review]

**Pull request merged**